cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -192,10 +206,15 @@ void reader::impl::decode_page_data(read_mode mode, size_t skip_rows, size_t num
       if (owning_schema == 0 || owning_schema == input_col.schema_idx) {
         valids[idx] = out_buf.null_mask();
         data[idx]   = out_buf.data();
+        // String size of the current column
+        auto const col_string_size = col_string_sizes[pass.chunks[c].src_col_index];
         // only do string buffer for leaf
-        if (idx == max_depth - 1 and out_buf.string_size() == 0 and
-            col_string_sizes[pass.chunks[c].src_col_index] > 0) {
-          out_buf.create_string_data(col_string_sizes[pass.chunks[c].src_col_index], _stream);
+        if (idx == max_depth - 1 and out_buf.string_size() == 0 and col_string_size > 0) {
+          out_buf.create_string_data(
+            col_string_size,
+            pass.cumulative_col_string_sizes[pass.chunks[c].src_col_index] >
+              static_cast<size_t>(strings::detail::get_offset64_threshold()),
```

Comment:
Does this mean, we may create some small-string columns for the first several chunks, then suddenly we see large string thus we will create all large string columns from that chunk until the end of the reading?

Reply:
Yes, that's because we use [str_offset](https://github.com/rapidsai/cudf/blob/0b9277b3abe014b9ab1cf7f849c36b21c2422bbe/cpp/src/io/parquet/parquet_gpu.hpp#L338) for pages within the same pass to compute string offsets in the output buffer, which along with `offptr` is typecasted to `cudf::size_type` here https://github.com/rapidsai/cudf/blob/0b9277b3abe014b9ab1cf7f849c36b21c2422bbe/cpp/src/io/parquet/page_string_decode.cu#L1137

That being said, with default chunking mechanism, we will encounter big n-1 chunks and only the last chunk to be smaller (remnant) depending on the `pass_read_limit` set.

Reply:
Is is possible to have the first `n` chunk having small-string columns while the following chunks have large-string columns? In addition, if having mixed small/large string columns in different chunks, will there be anything to be concerned?

Reply:
> Is it possible to have the first n chunk having small-string columns while the following chunks have large-string columns?
Very unlikely imo as it would mean having front-loaded non-string columns and back-loaded string cols. Regardless, we should be able to handle it with the changes in this PR keeping a track of the `offset_ptr` of all string cols.

> if having mixed small/large string columns in different chunks, will there be anything to be concerned?
No we compute string offsets (from a column's pages) per column so for any non-string cols, that will be zero